### ISSUE TITLE
TSDAE loss

### ISSUE THREAD

**Issue by rufimelo99** [UKPLab/sentence-transformers#1780]
I've been trying to perform domain adaptation on a BERT model using TSDAE.
How can I output the loss during the training steps?
Also, how can I evaluate it on a testing or validation dataset?

The code I am using is this one:
```
device_cuda = 0
epochs = 1
learning_rate = 3e-5

starting_steps = 0
delta_train_steps = 100
max_train_steps = 10000


model_name = 'neuralmind/bert-large-portuguese-cased'
model = SentenceTransformer(model_name)
torch.cuda.set_device(device_cuda)

from datasets import load_dataset
dataset = load_dataset('stjiris/portuguese-legal-sentences-v0')
df2 = pd.DataFrame(dataset['train'])
df2 = df2.dropna()



train_sentences = df2[steps:steps+delta_train_steps]['sentence'].values.tolist()

# Create the special denoising dataset that adds noise on-the-fly
train_dataset = datasets.DenoisingAutoEncoderDataset(train_sentences)

# DataLoader to batch your data
train_dataloader = DataLoader(train_dataset, batch_size=4, shuffle=True)

# Use the denoising auto-encoder loss
train_loss = losses.DenoisingAutoEncoderLoss(model, decoder_name_or_path=model_name, tie_encoder_decoder=True)

device = torch.device("cuda:"+str(device_cuda) if torch.cuda.is_available() else "cpu")
model.to(device)

    
# Call the fit method
model.fit(
    train_objectives=[(train_dataloader, train_loss)],
    epochs=epochs,
    weight_decay=0,
    scheduler='constantlr',
    optimizer_params={'lr': learning_rate},
    show_progress_bar=True
)
```

**Comment by MattiaSangermano** [comment]
I think the best way to evaluate a model is by using an evaluator. If you want you can choose one from `sentence_transformers.evaluation`. The code would be like that:

```python
from sentence_transformers.evaluation import TripletEvaluator

.
.
.

# Define the evaluator
validation_evaluation = TripletEvaluator(val_anchor, val_positives, val_negatives)

# Call the fit method
model.fit(
    train_objectives=[(train_dataloader, train_loss)],
    epochs=epochs,
    weight_decay=0,
    scheduler='constantlr',
    optimizer_params={'lr': learning_rate},
    show_progress_bar=True,
    evaluator=validation_evaluation
)
```

In addition, if you set `save_best_model=True` in the fit function, the model having the best score will be saved in the `output_path`.

Regarding the logging of the loss, according to #1056 there is no way to track it yet.

**Comment by lishuaiCN** [comment]
#336 maybe it's useful for you

**Comment by rufimelo99** [comment]
Thanks for the tip, comments on issue #336 provides an implementation for this matter.
